### PR TITLE
 Write entire buffer to disk rather than per frame

### PR DESCRIPTION
**Pull request by flynnmd**
Speeds up writing the processed audio to disk by quite a lot, processing a ~10m file goes from taking ~50s to ~10s.

A better solution would to write in chunks of frames rather than the entire signal since this requires a large heap allocation, but this is simple and works. Let me know if you'd rather have that though.

Also not sure about re-adding the Makefile from master since I'm not sure how this branch wants to be built, but this seems to work as well.

### CODE REVIEW

File: noisereduction/NoiseReduction.cpp [hunk 1]
```diff
@@ -980,18 +980,18 @@ void NoiseReduction::ReduceNoise(const char* outputPath, size_t t0, size_t t1) {
...
     LOG_F(INFO, "Writing %zd frames to disk", frameCount);
 
+    // write audio to buffer so that the channels are interleaved
+    float* buffer = new float[frameCount*mCtx.info.channels];
     for (int i = 0; i < frameCount; i++) {
-        // create frame
-        float buffer[mCtx.info.channels];
         for (int currentChannel = 0; currentChannel < channels; currentChannel++) {
-            buffer[currentChannel] = outputs[currentChannel].data[i];
+            buffer[i*mCtx.info.channels+currentChannel] = outputs[currentChannel].data[i];
```
Comment (tals) [inline noisereduction/NoiseReduction.cpp:987]: nit: operator spacing

File: noisereduction/NoiseReduction.cpp [hunk 2]
```diff
@@ -980,18 +980,18 @@ void NoiseReduction::ReduceNoise(const char* outputPath, size_t t0, size_t t1) {
     auto frameCount = outputs[0].length;
     LOG_F(INFO, "Writing %zd frames to disk", frameCount);
 
+    // write audio to buffer so that the channels are interleaved
+    float* buffer = new float[frameCount*mCtx.info.channels];
```
Comment (tals) [inline noisereduction/NoiseReduction.cpp:984]: nit: std::unique_ptr

**Comment by tals** [comment]
The makefile was added after I branched this off. Let me rebase :)

**Comment by flynnmd** [comment]
Going to close this and make a new PR since the `features/optimization` branch actually was behind master :P

**Pull request closed without merging**